server/src/core/server/graph/schema/schema.graphql

```diff
@@ -3153,6 +3153,27 @@ type User {
   bio is a user-defined biography.
   """
   bio: String
+
+  """
+  hasNewNotifications returns true if the user has received new notifications
+  since they last viewed their notifications tab.
+  """
+  hasNewNotifications: Boolean!
+    @auth(
+      userIDField: "id"
+      roles: [ADMIN, MODERATOR]
```

Comment:
Just curious, why is this (and `lastSeenNotificationDate`) restricted to roles admin and moderator?

Reply:
I believe this is saying that `hasNewNotifications` is restricted to the viewer's own notifications (by userID match) and can also be accessed by roles `ADMIN` and `MODERATOR`.

I have a commenter user who is not an admin or a moderator and because of the `userIDField` they are able to query their own notifications.

Reply:
Ah okay, got it...thank you!